### DISCUSSION TITLE
Dialogue narrator breaks

### DISCUSSION THREAD

**Discussion by Nauthizz** [vkbo/novelWriter#1976] [Help]
I'm posting this in discussions first, beacue I'm not sure if it is a bug or maybe just a simple case of PEBCAK. 

I've been trying to get the dialogue highlighting to work with Polish punctuation with mixed results. Only dashes are used (either m dash or n dash) with spaces on both sides (no space at the beginning of the line, though). Quotes are not used for dialogue at all.

I used n dash (quicker to type) and set it both as a line symbol and narrator break symbol, but this way the whole paragraph is highlighted and no breaks recognized:

![dialogue-highlighing-pref](https://github.com/user-attachments/assets/61c93e8a-d75e-43d4-ad99-24ef3a74810d)
![dialogue-highlighing-01](https://github.com/user-attachments/assets/9f86f61e-3f64-4998-ac71-9aa71269fabf)

I also tried to paste a sequence of space-dash-space for narrator break, but then the highlighting went all wonky, marking text after various special characters as a dialogue. Is it even possible for this option to take more than one character?

Here's what I'd like to see (hopefully):
![dialogue-highlighing-02](https://github.com/user-attachments/assets/e7fd0ece-5132-4362-b04d-7bd733f7dca0)

The text from my sceeenshots, translated verbatim:

```
This is what an example dialogue might look like using Polish punctuation rules:

– Example statement – someone said. And he added: – Another example statement.

– Oh my! – It would be nice if only the statements were highlighted, without any narration. In a paragraph where there is only a short statement and then a lot happens, this would be especially justified.

There are also sometimes paragraphs that start with a narrative, and only then someone shouts out the words: – Oooh! Look!

And so on and so forth. However, "text in quotation marks" should not be highlighted at all, and if so, it should be highlighted differently.


```

**Comment by vkbo** [comment]
The narrator break does not accept spaces on the "narrator" side of the text. The rule was made for Spanish, which I was told shouldn't have a space. See https://github.com/vkbo/novelWriter/issues/1771#issuecomment-2015894973

The feature was added only as requested, and my idea was to extend it based on user feedback.

If padding should be allowed, I need to add a specific switch for that since allowing it by default may break other functionality.

> Is it even possible for this option to take more than one character?

The input box should only accept a single character, so not sure how you even managed to get more 😄

**Reply by vkbo** [reply]
As for the format you'd like to see, does it mean that for every – you turn dialogue on and off? That will not work here. If we allow spaces around the symbol, it would be very confused in dialogue that has more than one narrator break. That's why the format doesn't allow spaces in the first place.

I think we would need to come up with a very different parsing for Polish if that's the case.

**Reply by Nauthizz** [reply]
> If padding should be allowed, I need to add a specific switch for that since allowing it by default may break other functionality.

Should I post it as a separate issue then?

> The input box should only accept a single character, so not sure how you even managed to get more 😄

Ctrl+C Ctrl+V, obviously 😄 But I guess it only accepted the first space and this is what made it act so weirdly 😄

**Reply by vkbo** [reply]
> Should I post it as a separate issue then?

Sure, that would be helpful. I couldn't find anywhere online that describes the style used.

> Ctrl+C Ctrl+V, obviously 😄 But I guess it only accepted the first space and this is what made it act so weirdly 😄

Yes, that would leave just a space, and that would probably make it behave strangely. Preferences should probably not accept spaces in those boxes at all.

**Comment by vkbo** [comment]
Ok, so I've solved most of your use cases in #2068.

![image](https://github.com/user-attachments/assets/e08f9814-f212-437d-9dab-d595d2f7b113)

The last one, with dialogue in a paragraph that starts with narration is trickier, I can add it, but that means you cannot use the dash as a pause in the text anywhere as it too would be interpreted as dialogue.

**Reply by vkbo** [reply]
I've made another update that should cover all your cases as well. See this comment and the following one: https://github.com/vkbo/novelWriter/issues/1773#issuecomment-2450897638
